test/unit/slurm/test_slurm_config_generator/test_generate_slurm_config_with_custom_settings/sample_instance_types_data.json

```diff
@@ -0,0 +1 @@
 [1, 2]}, "MemoryInfo": {"SizeInMiB": 249856}, "InstanceStorageSupported": false, "EbsInfo": {"EbsOptimizedSupport": "default", "EncryptionSupport": "supported", "EbsOptimizedInfo": {"BaselineBandwidthInMbps": 7000, "BaselineThroughputInMBps": 875.0, "BaselineIops": 40000, "MaximumBandwidthInMbps": 7000, "MaximumThroughputInMBps": 875.0, "MaximumIops": 40000}, "NvmeSupport": "unsupported"}, "NetworkInfo": {"NetworkPerformance": "10 Gigabit", "MaximumNetworkInterfaces": 8, "MaximumNetworkCards": 1, "DefaultNetworkCardIndex": 0, "NetworkCards": [{"NetworkCardIndex": 0, "NetworkPerformance": "10 Gigabit", "MaximumNetworkInterfaces": 8}], "Ipv4AddressesPerInterface": 30, "Ipv6AddressesPerInterface": 30, "Ipv6Supported": true, "EnaSupport": "supported", "EfaSupported": false}, "GpuInfo": {"Gpus": [{"Name": "M60", "Manufacturer": "NVIDIA", "Count": 2, "MemoryInfo": {"SizeInMiB": 8192}}], "TotalGpuMemoryInMiB": 16384}, "PlacementGroupInfo": {"SupportedStrategies": ["cluster", "partition", "spread"]}, "HibernationSupported": false, "BurstablePerformanceSupported": false, "DedicatedHostsSupported": true, "AutoRecoverySupported": false}, "p3dn.24xlarge": {"InstanceType": "p3dn.24xlarge", "CurrentGeneration": true, "FreeTierEligible": false, "SupportedUsageClasses": ["spot", "ondemand"], "SupportedRootDeviceTypes": ["ebs"], "SupportedVirtualizationTypes": ["hvm"], "BareMetal": false, "Hypervisor": "nitro", "ProcessorInfo": {"SupportedArchitectures": ["x86_64"], "SustainedClockSpeedInGhz": 2.5}, "VCpuInfo": {"DefaultVCpus": 96, "DefaultCores": 48, "DefaultThreadsPerCore": 2, "ValidCores": [4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48], "ValidThreadsPerCore": [1, 2]}, "MemoryInfo": {"SizeInMiB": 786432}, "InstanceStorageSupported": true, "InstanceStorageInfo": {"TotalSizeInGB": 1800, "Disks": [{"SizeInGB": 900, "Count": 2, "Type": "ssd"}], "NvmeSupport": "required"}, "EbsInfo": {"EbsOptimizedSupport": "default", "EncryptionSupport": "supported", "EbsOptimizedInfo": {"BaselineBandwidthInMbps": 19000, "BaselineThroughputInMBps": 2375.0, "BaselineIops": 80000, "MaximumBandwidthInMbps": 19000, "MaximumThroughputInMBps": 2375.0, "MaximumIops": 80000}, "NvmeSupport": "required"}, "NetworkInfo": {"NetworkPerformance": "100 Gigabit", "MaximumNetworkInterfaces": 15, "MaximumNetworkCards": 1, "DefaultNetworkCardIndex": 0, "NetworkCards": [{"NetworkCardIndex": 0, "NetworkPerformance": "100 Gigabit", "MaximumNetworkInterfaces": 15}], "Ipv4AddressesPerInterface": 50, "Ipv6AddressesPerInterface": 50, "Ipv6Supported": true, "EnaSupport": "required", "EfaSupported": true}, "GpuInfo": {"Gpus": [{"Name": "V100", "Manufacturer": "NVIDIA", "Count": 8, "MemoryInfo": {"SizeInMiB": 32768}}], "TotalGpuMemoryInMiB": 262144}, "PlacementGroupInfo": {"SupportedStrategies": ["cluster", "partition", "spread"]}, "HibernationSupported": false, "BurstablePerformanceSupported": false, "DedicatedHostsSupported": false, "AutoRecoverySupported": false}}
```

Comment:
Let's keep here only the instance types we are going to keep in the minimal test yaml file.

Also watch out, this file does not end with a newline character.

Reply:
I copied the file from another test so I assume it is valid.

We should reuse the same file across multiple test instead of copying it around, but this is out-of-scope for this PR.